### PR TITLE
feat: detect vercel env

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 6a87be7</samp>

Added console logging of environment variables in `MainFeed.tsx` to debug Vercel deployment issues.

### <samp>🤖 Generated by Copilot at 6a87be7</samp>

> _`VERCEL_ENV` logged_
> _To solve deployment issues_
> _Autumn leaves falling_

### <samp>🤖 Generated by Copilot at 6a87be7</samp>

*  Added code to log environment variables in the console ([link](https://github.com/Crossbell-Box/xLog/pull/425/files?diff=unified&w=0#diff-6af61e4e539bc42a885190d59c840a2e6426bc4aea2d66a49e2cad0427826f76R45-R50))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Comment by Innei** [comment]
@DIYgod  It looks `Next.js Bundle Analysis / analyze (pull_request) ` CI still has a problem. In answer to the previous question, my understanding is this vercel-bot is a globally installed GitHub App that was previously granted permissions and is not triggered by workflow CI (in fact it should be triggered by webhooks), So CI triggers require create-comment or update-comment permissions.

File: src/lib/helpers.ts [hunk 1]
```diff
@@ -14,15 +14,31 @@ export const getSiteLink = ({
...
   }
   if (noProtocol) {
     return `${subdomain}.${OUR_DOMAIN}`
   }
+
   return `${IS_PROD ? "https" : "http"}://${subdomain}.${OUR_DOMAIN}`
 }
 
+export const getSlugUrl = (slug: string) => {
+  if (IS_VERCEL_PREVIEW || IS_DEV) {
```
Comment (DIYgod) [inline src/lib/helpers.ts:30]: Why do we need it in the development environment? Do you also have subdomain issues during the development process?

File: src/lib/helpers.ts [hunk 2]
```diff
@@ -14,15 +14,31 @@ export const getSiteLink = ({
...
+    const pathArr = new URL(location.href).pathname.split("/").filter(($) => $)
+    const indicatorIndex = pathArr.findIndex(($) => $ === "_site")
+    if (-~indicatorIndex) {
```
Comment (DIYgod) [inline src/lib/helpers.ts:33]: What is this magic syntax 🤯

Comment (Innei) [reply src/lib/helpers.ts:34]:
```
-~-1 => -0
-~0 => 1
```

判断不是 -1 的 case

File: src/lib/helpers.ts (hunk 1 above)
Comment (Innei) [reply src/lib/helpers.ts]: I will remove it

Comment (Innei) [reply src/lib/helpers.ts]: I will remove it

**Pull request merged**